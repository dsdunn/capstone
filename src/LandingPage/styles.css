* {
  font-family: 'Roboto', sans-serif; }

.wrapper {
  /* The height needs to be set to a fixed value for the effect to work.
   * 100vh is the full height of the viewport. */
  height: 100vh;
  width: 100vw;
  /* The scaling of the images would add a horizontal scrollbar, so disable x overflow. */
  overflow-x: hidden;
  /* Enable scrolling on the page. */
  overflow-y: auto;
  /* Set the perspective to 2px. This is essentailly the simulated distance from the viewport to transformed objects.*/
  perspective: 3px; }

.section {
  /* Needed for children to be absolutely positioned relative to the parent. */
  position: relative;
  /* The height of the container. Must be set, but it doesn't really matter what the value is. */
  height: 50%;
  /* For text formatting. */
  display: flex;
  align-items: center;
  justify-content: center;
  color: white; }

h1 {
  font-size: 80px; }

.project-name {
  font-size: 120px; }

.parallax::after {
  /* Display and position the pseudo-element */
  content: " ";
  position: absolute;
  top: -15vh;
  right: 0;
  bottom: 0;
  left: 0;
  /* Move the pseudo-element back away from the camera,
   * then scale it back up to fill the viewport.
   * Because the pseudo-element is further away, it appears to move more slowly, like in real life. */
  transform: translateZ(-1px) scale(1.5);
  /* Force the background image to fill the whole element. */
  background-size: 100%;
  /* Keep the image from overlapping sibling elements. */
  z-index: -1; }

/* The styling for the static div. */
.static {
  height: 450px;
  box-shadow: 0px 0px 10px black;
  background: black; }

.bg1 {
  height: 55vh; }

.bg1::after {
  background-image: url("../images/1280-jp-record-shelf-1.jpg"); }

.bg2::after {
  background-image: url("https://images.pexels.com/photos/264544/pexels-photo-264544.jpeg?auto=compress&cs=tinysrgb&dpr=2&h=750&w=1260"); }

.bg3::after {
  background-image: url("http://sports.mearsonlineauctions.com/ItemImages/000076/76208a_lg.jpeg"); }

.bg4::after {
  background-image: url("http://mediad.publicbroadcasting.net/p/wglt/files/styles/x_large/public/201603/160318ComicBooks_0.jpg"); }

.bg5::after {
  background-image: url("http://www.notwedordead.com/wp-content/uploads/2015/05/My-postcard-collection-4d5f267c2d94c_hires.jpg"); }

.s1 {
  background-color: black; }

.s3 {
  background-color: black; }
