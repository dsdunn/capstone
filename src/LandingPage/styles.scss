* {
  font-family: 'Roboto', sans-serif;
}



.wrapper {
  /* The height needs to be set to a fixed value for the effect to work.
   * 100vh is the full height of the viewport. */
  height: 100vh;
  width: 100vw;
  /* The scaling of the images would add a horizontal scrollbar, so disable x overflow. */
  overflow-x: hidden;
  /* Enable scrolling on the page. */
  overflow-y: auto;
  /* Set the perspective to 2px. This is essentailly the simulated distance from the viewport to transformed objects.*/
  perspective: 3px;
}

.section {
  /* Needed for children to be absolutely positioned relative to the parent. */
  position: relative;
  /* The height of the container. Must be set, but it doesn't really matter what the value is. */
  height: 50%;

  /* For text formatting. */
  display: flex;
  align-items: center;
  justify-content: center;
  color: white;
}


h1 {
  font-size: 80px
}

.project-name {
  font-size: 120px;
}

.parallax::after {
  /* Display and position the pseudo-element */
  content: " ";
  position: absolute;
  height: 60vh;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;

  /* Move the pseudo-element back away from the camera,
   * then scale it back up to fill the viewport.
   * Because the pseudo-element is further away, it appears to move more slowly, like in real life. */
  transform: translateZ(-1px) scale(1.5);
  /* Force the background image to fill the whole element. */
  background-size: 100%;
  /* Keep the image from overlapping sibling elements. */
  z-index: -1;
}

/* The styling for the static div. */
.static {
  height: 40vh;
  box-shadow: 0px 0px 10px black;
  background: #262425 ;
}

.bg1 {
  height:65vh;
}


.bg1::after {
  background-image: url('../images/vinyl.jpg');
}

// https://images.pexels.com/photos/157519/pexels-photo-157519.jpeg?auto=compress&cs=tinysrgb&dpr=2&h=750&w=1260

.bg2::after {
  background-image: url('http://mediad.publicbroadcasting.net/p/wglt/files/styles/x_large/public/201603/160318ComicBooks_0.jpg');
}

.bg3::after {
  background-image: url('http://sports.mearsonlineauctions.com/ItemImages/000076/76208a_lg.jpeg');
}

.bg4::after {
  background-image: url('../images/highrescoincol.jpg');
  background-position: bottom;
}

// .bg5::after {
//   background-image: url('../images/highrescoincol.jpg');
// }

.s1 {
  background-color: #aa2c1b;
}

.s2 {
  background-color: #349164;
}

.s3 {
  background-color: #2e74aa;
}
