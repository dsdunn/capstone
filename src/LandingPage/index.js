import React from 'react';
import { Link } from 'react-router-dom';
import './styles.css'

const LandingPage = () => {
  return (
    <div class="wrapper">
      <section className="section parallax bg1">
        <h1 className='project-name'>Collecshare</h1>
      </section>
      <section className="section static s1">
        <h1>Collect.</h1>
      </section>
      <section className="section parallax bg2"></section>
      <section className="section static s2">
        <h1>Share.</h1>
      </section>
      <section className="section parallax bg3"></section>
      <section className="section static s3">
        <h1>Connect.</h1>
      </section>
      <section className="section parallax bg4"></section>
      <section className="section static s4">
        <h1>Explore.</h1>
      </section>
      <section className="section parallax bg5"></section>
    </div>
  )
}

export default LandingPage;