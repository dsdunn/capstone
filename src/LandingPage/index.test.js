import React from 'react';
import { shallow } from 'enzyme';
import LandingPage from './'

describe('LandingPage', () => {
  const wrapper = shallow(<LandingPage />)

  it('should match the snapShot', () => {
    expect(wrapper).toMatchSnapshot()
  })
})