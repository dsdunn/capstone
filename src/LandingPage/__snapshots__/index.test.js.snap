// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingPage should match the snapShot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LandingPage />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <section
          className="section parallax bg1"
>
          <h1
                    className="project-name"
          >
                    Collecshare
          </h1>
</section>,
        <section
          className="section static s1"
>
          <h1>
                    Collect.
          </h1>
</section>,
        <section
          className="section parallax bg2"
/>,
        <section
          className="section static s2"
>
          <h1>
                    Share.
          </h1>
</section>,
        <section
          className="section parallax bg3"
/>,
        <section
          className="section static s3"
>
          <h1>
                    Connect.
          </h1>
</section>,
        <section
          className="section parallax bg4"
/>,
        <section
          className="section static s4"
>
          <h1>
                    Explore.
          </h1>
</section>,
      ],
      "className": "wrapper",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <h1
            className="project-name"
>
            Collecshare
</h1>,
          "className": "section parallax bg1",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Collecshare",
            "className": "project-name",
          },
          "ref": null,
          "rendered": "Collecshare",
          "type": "h1",
        },
        "type": "section",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <h1>
            Collect.
</h1>,
          "className": "section static s1",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Collect.",
          },
          "ref": null,
          "rendered": "Collect.",
          "type": "h1",
        },
        "type": "section",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "section parallax bg2",
        },
        "ref": null,
        "rendered": null,
        "type": "section",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <h1>
            Share.
</h1>,
          "className": "section static s2",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Share.",
          },
          "ref": null,
          "rendered": "Share.",
          "type": "h1",
        },
        "type": "section",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "section parallax bg3",
        },
        "ref": null,
        "rendered": null,
        "type": "section",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <h1>
            Connect.
</h1>,
          "className": "section static s3",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Connect.",
          },
          "ref": null,
          "rendered": "Connect.",
          "type": "h1",
        },
        "type": "section",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "section parallax bg4",
        },
        "ref": null,
        "rendered": null,
        "type": "section",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <h1>
            Explore.
</h1>,
          "className": "section static s4",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Explore.",
          },
          "ref": null,
          "rendered": "Explore.",
          "type": "h1",
        },
        "type": "section",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <section
            className="section parallax bg1"
>
            <h1
                        className="project-name"
            >
                        Collecshare
            </h1>
</section>,
          <section
            className="section static s1"
>
            <h1>
                        Collect.
            </h1>
</section>,
          <section
            className="section parallax bg2"
/>,
          <section
            className="section static s2"
>
            <h1>
                        Share.
            </h1>
</section>,
          <section
            className="section parallax bg3"
/>,
          <section
            className="section static s3"
>
            <h1>
                        Connect.
            </h1>
</section>,
          <section
            className="section parallax bg4"
/>,
          <section
            className="section static s4"
>
            <h1>
                        Explore.
            </h1>
</section>,
        ],
        "className": "wrapper",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h1
              className="project-name"
>
              Collecshare
</h1>,
            "className": "section parallax bg1",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Collecshare",
              "className": "project-name",
            },
            "ref": null,
            "rendered": "Collecshare",
            "type": "h1",
          },
          "type": "section",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h1>
              Collect.
</h1>,
            "className": "section static s1",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Collect.",
            },
            "ref": null,
            "rendered": "Collect.",
            "type": "h1",
          },
          "type": "section",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "section parallax bg2",
          },
          "ref": null,
          "rendered": null,
          "type": "section",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h1>
              Share.
</h1>,
            "className": "section static s2",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Share.",
            },
            "ref": null,
            "rendered": "Share.",
            "type": "h1",
          },
          "type": "section",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "section parallax bg3",
          },
          "ref": null,
          "rendered": null,
          "type": "section",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h1>
              Connect.
</h1>,
            "className": "section static s3",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Connect.",
            },
            "ref": null,
            "rendered": "Connect.",
            "type": "h1",
          },
          "type": "section",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "section parallax bg4",
          },
          "ref": null,
          "rendered": null,
          "type": "section",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h1>
              Explore.
</h1>,
            "className": "section static s4",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Explore.",
            },
            "ref": null,
            "rendered": "Explore.",
            "type": "h1",
          },
          "type": "section",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
